org.eclipse.wildwebdeveloper/pom.xml

```diff
@@ -93,9 +93,19 @@
 							<skipCache>true</skipCache>
 							<url>https://raw.githubusercontent.com/microsoft/vscode-eslint/release/2.2.2/server/package.json</url>
 							<outputDirectory>${project.build.directory}/vscode-eslint-ls/extension/server</outputDirectory>
-							<headers>
-								<Authorization>${github.api.token}</Authorization>
-							</headers>
+							<if>
+								<condition>
+									<not>
+										<equals arg1="${github.api.token}" arg2="" />
+									</not>
+								</condition>
+								<then>
+									<headers>
+										<Authorization>token ${github.api.token}</Authorization>
+									</headers>
+								</then>
+							</if>
+
```

Comment:
How is it supposed to work? The wget mojo doesn't support `<if>` as far as I know, thos this whole block may be useless.

Reply:
when we don't provide a value for `${github.api.token}` (running it manually with `mvn clean verify`, f.i.,  or in WindowsPR build) this construction just doesn't add the `<headers>...</headers>` fragment to the effective pom.xml (so it's omitted).
When `${github.api.token}` is set to some value (like in Jenkins build) then `<headers>...</headers>` fragment is added to the effective pom.xml and that value is used in the requests header.

... or something like this.

Reply:
What makes you think this actually works? I'm not aware that `<if>` is a native Maven construct, nor that download-maven-plugin has support for this? The standard way to conditionally configure a pom.xml is to use Maven profiles.

Reply:
The example is taken from stackoverflow: https://stackoverflow.com/questions/10149539/conditional-configuration-in-maven-pom-xml
From the experience - we have two build configurations - WindowsPR and Jenkinsfile, also we can build locally running `mvn clean install`. So, after we added the github token as a parameter it become a required paramerter - as result local build with `mvn clean install` and WindowsPR build had started to fail (because we don't add this parameter to the commands. Jenkinsfile build started successfully download the required `package.json` file from `raw.githubuser...`. 
Adding the following `<if><condition>...<then>` construction:
```
							<if>
								<condition>
									<not>
										<equals arg1="${github.api.token}" arg2="" />
									</not>
								</condition>
								<then>
									<headers>
										<Authorization>token ${github.api.token}</Authorization>
									</headers>
								</then>
							</if>
```
made the local and WindowsPR builds to work normally.
So, I suppose it really works. 
If no `github.api.token` parameter value is provided, the `<headers>...</headers>` elements are not added to the effective pom.xml and, as result, wget action doesn't fail because of wrong header is sent (a default header is sent in this case)